pc-relate/pcrelate.wdl

```diff
@@ -0,0 +1,395 @@
+		if "~{kinship_method}" not in ['pcrelate', 'king_ibdseg', 'king_robust']:
+			f.close()
+			print("Invalid kinship method. Must be pcrelate, king_robust, or king_ibdseg")
+			exit(1)
+		elif "~{kinship_method}" == "king_robust":
+			f.write('kinship_method "king"\n')
+		else:
+			f.write('kinship_method "~{kinship_method}"\n')
+
+		if "~{kinship_plot_threshold}" is not "":
+			f.write('kinship_threshold "~{kinship_plot_threshold}"\n')
+
+		f.write('out_file_all "~{out_prefix_final}_all.pdf"\n')
+		f.write('out_file_cross "~{out_prefix_final}_cross_group.pdf"\n')
+		f.write('out_file_study "~{out_prefix_final}_within_group.pdf"\n')
+
+		if "~{phenotype_file}" is not "":
+			f.write('phenotype_file "~{phenotype_file}"\n')
+		
+		if "~{group}" is not "":
+			f.write('study "~{group}"\n')
+
+		if "~{sample_include_file}" is not "":
+			f.write('sample_include_file "~{sample_include_file}"\n')
+
+		CODE
+
+		R -q --vanilla --args kinship_plots.config < /usr/local/analysis_pipeline/R/kinship_plots.R
+	}
+	
+	# Estimate disk size required
+	Int kin_size = ceil(size(kinship_file, "GB"))
+	Int phen_size = select_first([ceil(size(phenotype_file, "GB")), 0])
+	Int smpl_size = select_first([ceil(size(sample_include_file, "GB")), 0])
+	Int finalDiskSize = kin_size + phen_size + smpl_size + addldisk
+	
+	runtime {
+		cpu: cpu
+		docker: "uwgac/topmed-master@sha256:0bb7f98d6b9182d4e4a6b82c98c04a244d766707875ddfd8a48005a9f5c5481e"
+		disks: "local-disk " + finalDiskSize + " HDD"
+		memory: "${memory} GB"
+		preemptibles: "${preempt}"
+	}
+	output {
+		Array[File] kinship_plots = glob("*.pdf")
+	}
+}
+
+workflow pcrel {
+	input {
+		File pca_file
+		File gds_file
+		File? sample_include_file
+		Int? n_pcs # sb visual default is 3, check if gets passed in as such
+		File? variant_include_file
+		Int? variant_block_size
+		String? out_prefix
+		Int? n_sample_blocks # sb default 1
+		File? phenotype_file
+		Float? kinship_plot_threshold
+		String? group
+		Float? sparse_threshold
+		Boolean ibd_probs = true
+	}
+
+	call pcrelate_beta {
+		input:
+			gds_file = gds_file,
+			pca_file = pca_file,
+			n_pcs = n_pcs,
+			out_prefix = out_prefix,
+			sample_include_file = sample_include_file,
+			variant_include_file = variant_include_file,
+			variant_block_size = variant_block_size
+	}
+
+	call sample_blocks_to_segments {
+		input:
+			n_sample_blocks = n_sample_blocks
+	}
+
+	scatter(seg in sample_blocks_to_segments.segments) {
+		call pcrelate {
+			input:
+				gds_file = gds_file,
+				pca_file = pca_file,
+				beta_file =pcrelate_beta.beta,
+				n_pcs = n_pcs,
+				out_prefix = out_prefix,
+				variant_include_file = variant_include_file,
+				variant_block_size = variant_block_size,
+				sample_include_file = sample_include_file,
+				n_sample_blocks = n_sample_blocks,
+				segment = seg,
+				ibd_probs = ibd_probs
+		}
+	}
+
+	call pcrelate_correct {
+		input:
+			n_sample_blocks = n_sample_blocks,
+			pcrelate_block_files = pcrelate.block,
```

Comment:
The input should be pcrelate.pcrelate

Reply:
I'm wondering if there was a problem with conflicting names?

Reply:
Turns out it actually _is_ valid. I must've set it to blocks just to avoid confusing myself. Changed it back to pcrelate.